prometheusbp/promtest/spec.go

```diff
@@ -0,0 +1,80 @@
+package promtest
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+var (
+	errPrefix         = errors.New("the prefix is not at the beginning of the metric name")
+	errLength         = errors.New("metric name should have a minimum of 3 parts, like prefix_name_suffix")
+	errCount          = errors.New("wrong metric count for prefix")
+	errPrometheusLint = errors.New("problem with Prometheus GatherAndLint")
+)
+
+// ValidateSpec validates that the Prometheus metrics being exposed conform to
+// the baseplate spec.
+func ValidateSpec(t *testing.T, metricPrefix string, wantMetricCount int) {
+	if err := validateSpec(metricPrefix, wantMetricCount); err != nil {
+		t.Error(err)
+	}
+}
+
+func validateSpec(metricPrefix string, wantMetricCount int) error {
+	metricFam, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		return err
+	}
+
+	var gotMetricCount int
+	for _, m := range metricFam {
+		metricName := *m.Name
+
+		if strings.Contains(metricName, metricPrefix) {
```

Comment:
* Use `HasPrefix` not `Contains`

* Reduce indentation by continuing early

```go
if !strings.HasPrefix(metricName, metricPrefix) {
  continue
}
```

Reply:
I chose to use `Contains` instead of `HasPrefix` to catch if the  intended prefix ended up in a different part of the metric name... but as I write that now maybe its an unnecessary additional step.